core/meta/src/TClass.cxx

```diff
@@ -7061,7 +7061,7 @@ TVirtualStreamerInfo *TClass::GetConversionStreamerInfo( const TClass* cl, Int_t
          arr = it->second;
       }
 
-      if( arr && version > -1 && version < arr->GetSize() && arr->At( version ) )
+      if( arr && version >= -1 && version < arr->GetSize() && arr->At( version ) )
```

Comment:
What is the effect of having an entry at `-1` on `GetSize()`? I.e. `version < arr->GetSize()` might now not do what the original code expected, with `GetSize()` returning 1 with an element at index `-1`, yet for `version == 0` no element `arr->At(0)` exists despite `version < arr->GetSize()`.

Reply:
The array has an effective offset of +1, so the "real" indices are from [0 to max]. 

The array is not expected to be contiguous filled (in practice it is in most case filled in a couple of place, current version number and whatever old version was read).

For versioned class, the current version is stored both at -1 and 1.  Slot 0 is rarely used/filled as 0 indicates a class with all data members marked transient (so it happens only if such a class is used as base class).

> What is the effect of having an entry at -1 on GetSize()

it increases the size by 1 if the slot was previously empty.